emmet-core/emmet/core/electrode.py

```diff
@@ -0,0 +1,136 @@
+""" Core definition of an Electrode Document """
+from typing import List, Dict, ClassVar
+from datetime import datetime
+
+from pydantic import BaseModel, Field
+
+from emmet.stubs import ComputedEntry
+from emmet.core.structure import StructureMetadata
+
+
+class ElementEvolution(BaseModel):
+    """
+    Result of the PhaseDiagram.get element_profile which moves a along a line in
+    composition space and return the reaction, chemical potential, and the remaining
+    amount of the element
+    """
+
+    reaction: str = Field(
+        ...,
+        description="Evolution reaction determined by a facet of the phase diagram",
+    )
+
+    chempot: float = Field(
+        ...,
+        description="The chemical potential of the target element solved on that facet",
+    )
+
+    evolution: float = Field(
+        ..., description="Remaining amount of element .. TODO get better description",
+    )
+
+
+class VoltagePair(BaseModel):
+    """
+    Based on AbstractVoltagePair from pymatgen
+    """
+
+    voltage: float = Field(
+        ..., description="Voltage of voltage pair",
+    )
+
+    mAh: float = Field(
+        ..., description="Energy in mAh",
+    )
+
+    mass_charge: float = Field(
+        ..., description="Mass of charged pair.",
```

Comment:
I think you mean "charged end"? pair is used sorta ambiguously in these descriptions. It refers to both end members so these properties are for one of those two not the pair?